src/components/Ui/Select.vue

```diff
@@ -1,22 +1,24 @@
-<script setup lang="ts">
-type Item = {
-  key: string;
+<script setup lang="ts" generic="T extends string | number, U extends readonly Item<T>[]">
+export type Item<T extends string | number> = {
+  key: T;
   label: string;
   indicator?: string;
 };
 
 const props = defineProps<{
-  modelValue: string;
-  items: Item[];
+  modelValue: U[number]['key'];
+  title: string;
+  items: U;
   gap?: string;
   placement?: 'left' | 'right';
 }>();
 
 const emit = defineEmits<{
-  (e: 'update:modelValue', value: string): void;
+  (e: 'update:modelValue', value: U[number]['key']): void;
 }>();
 
 const currentItem = computed(() => props.items.find(item => item.key === props.modelValue));
+const items = computed(() => props.items);
```

Comment:
This doesn't seem to be needed. We can simply use `props.items` or `items` in template

Reply:
Without this TS types in template get crazy for some reason, looks like some edgecase/bug in Vue when using generics.

```bash
yarn run v1.22.19
$ vue-tsc --noEmit
src/components/Ui/Select.vue:45:56 - error TS2339: Property 'key' does not exist on type 'NonNullable<[{ type: PropType<U>; required: true; }] extends [Prop<infer V, infer D>] ? unknown extends V ? IfAny<V, V, D> : V : { type: PropType<U>; required: true; }>[keyof NonNullable<...>]'.

45       <UiDropdownItem v-for="item in items" :key="item.key" v-slot="{ active, disabled }">
                                                          ~~~

src/components/Ui/Select.vue:49:56 - error TS2339: Property 'key' does not exist on type 'NonNullable<[{ type: PropType<U>; required: true; }] extends [Prop<infer V, infer D>] ? unknown extends V ? IfAny<V, V, D> : V : { type: PropType<U>; required: true; }>[keyof NonNullable<...>]'.

49           @click="() => emit('update:modelValue', item.key)"
                                                          ~~~

src/components/Ui/Select.vue:52:24 - error TS2339: Property 'indicator' does not exist on type 'NonNullable<[{ type: PropType<U>; required: true; }] extends [Prop<infer V, infer D>] ? unknown extends V ? IfAny<V, V, D> : V : { type: PropType<U>; required: true; }>[keyof NonNullable<...>]'.

52             v-if="item.indicator"
                          ~~~~~~~~~

src/components/Ui/Select.vue:55:21 - error TS2339: Property 'indicator' does not exist on type 'NonNullable<[{ type: PropType<U>; required: true; }] extends [Prop<infer V, infer D>] ? unknown extends V ? IfAny<V, V, D> : V : { type: PropType<U>; required: true; }>[keyof NonNullable<...>]'.

55               [item.indicator]: true
                       ~~~~~~~~~

src/components/Ui/Select.vue:58:19 - error TS2339: Property 'label' does not exist on type 'NonNullable<[{ type: PropType<U>; required: true; }] extends [Prop<infer V, infer D>] ? unknown extends V ? IfAny<V, V, D> : V : { type: PropType<U>; required: true; }>[keyof NonNullable<...>]'.

58           {{ item.label }}
                     ~~~~~


Found 5 errors in the same file, starting at: src/components/Ui/Select.vue:45
```